assets/src/edit-story/app/rightClickMenu/provider.js

```diff
@@ -99,11 +126,66 @@ function RightClickMenuProvider({ children }) {
     ev.stopPropagation();
   }, []);
 
+  /**
+   * Copy the element and styles to the clipboard.
+   */
+  const handleCopyElement = useCallback(async () => {
+    if (!selectedElements.length) {
+      return;
+    }
+
+    const { htmlContent, serializedPayload } = serializeTextAndHTMLData(
+      currentPage,
+      selectedElements,
+      selectedElementAnimations
+    );
+
+    try {
+      await navigator.clipboard.writeText(
+        `<!-- ${serializedPayload} -->${htmlContent}`
+      );
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('trouble copying the dooby dooby doos', { error });
+    }
+  }, [currentPage, selectedElements, selectedElementAnimations]);
+
+  /**
+   * Paste the element and styles from the clipboard if one exists.
+   */
+  const handlePasteElement = useCallback(async () => {
+    let content;
+    try {
+      content = await navigator.clipboard.readText();
```

Comment:
Is it currently working ok for when mixing using keyboard and the right-click menu? E.g. if I copy using the keyboard and paste using the right-click menu?

Reply:
When I test it, it does! The right click menus aren't 100% done, so if I copy with a keyboard some text element and paste with the right click the text element is pasted but the styles aren't. This is fine since this menu is just to lay the initial menu structure with the page menu (which is the most "basic").

We are pivoting for 1.9 to focus entirely on the prepublish checklist and tabling the menus until 1.10. So the goal here is to reach a pause point that's easy to come back to the headspace of working on it.

Reply:
Sounds good.

Note: When I tested currently copying a shape with keyboard and then pasting with the right click, a new text element is created with the content "shape" instead of actually creating the shape -- I guess that's also part of a future iteration? Bringing it up just in case.

Reply:
> Sounds good.
> 
> Note: When I tested currently copying a shape with keyboard and then pasting with the right click, a new text element is created with the content "shape" instead of actually creating the shape -- I guess that's also part of a future iteration? Bringing it up just in case.

Good catch, this shouldn't happen but we have a ticket for the shape menu [here](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/6137). We'll fix it there!

Reply:
Great 👍

Reply:
Turns out this may all be good now. The copy/paste is supposed to work for the entire page. Just updated that so it doesn't copy/paste individual elements

Reply:
I see, so the right click menu is separate for page and the element. But if the user copies a shape from the element right click menu (or with keyboard) and then wants to paste it on an empty page using the right click menu, how would it work? Wouldn't the Paste option still need to paste a single element using the Page right-click menu? If pasting an element to a page is not possible from the Page right click menu, then it wouldn't be possible at all (using just the right click menu)?

Reply:
Great question and this is part of the reason why we're switching gears to focus on the checklist and then apply all our attention to these menus because they are sneaky in their complexity. 

This is just the first round of getting base functionality up and it's all behind a feature flag as it is still very much in flight.